server/claim/claim-revision/claim-revision.service.ts

```diff
@@ -204,17 +204,17 @@ export class ClaimRevisionService {
         if (sources && Array.isArray(sources)) {
             for (let source of sources) {
                 try {
-                    if (typeof source === "string") {
+                    const existingSource = await this.sourceService.findHref(source);
+                    if (existingSource && existingSource.length > 0) {
+                        for (let source of existingSource) {
```

Comment:
question: Since we can only have one source with a given href, wouldn't it be better to simplify the logic and remove the inner loop? Instead of iterating over existingSource, we can directly check for its existence and update it.